packages/pages/src/common/src/parsers/sourceFileParser.ts

```diff
@@ -164,19 +201,20 @@ export default class SourceFileParser {
     return this.sourceFile.getBaseName();
   }
 
+  /**
+   * Saves all changes made to source file.
+   */
   async save() {
     await this.sourceFile.save();
   }
-
-  private getAbsolutePath = (base: string, relative: string): string => {
-    const stack = base.split("/"),
-      parts = relative.split("/");
-    stack.pop();
-    for (let i = 0; i < parts.length; i++) {
-      if (parts[i] == ".") continue;
-      if (parts[i] == "..") stack.pop();
-      else stack.push(parts[i]);
-    }
-    return stack.join("/");
-  };
 }
+
+/**
+ * @param base the file path
+ * @param relative the relative file path
+ * @returns an absolute file path
+ */
+const getAbsolutePath = (base: string, relative: string): string => {
```

Comment:
This function is only used in one place and is relatively simply. Just combine it into one line in the place it's used.